glusterd2/commands/peers/addpeer.go

```diff
@@ -99,10 +99,18 @@ func addPeerHandler(w http.ResponseWriter, r *http.Request) {
 		newpeer.Metadata[key] = value
 	}
 
+	//check if remotePeerAddress already present
+	found := utils.StringInSlice(remotePeerAddress, newpeer.PeerAddresses)
+	//if not found prepend the remotePeerAddress to peer details
+	if !found {
+		newpeer.PeerAddresses = append([]string{remotePeerAddress}, newpeer.PeerAddresses...)
+	}
```

Comment:
if found, we can move that address as first address in the list?

Reply:
if not found, right ?

Reply:
@aravindavk,  done